asyncapi/hubspot/dispatcher_service.bal

```diff
@@ -0,0 +1,151 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerinax/asyncapi.native.handler;
+import ballerina/crypto;
+import ballerina/log;
+import ballerina/time;
+
+service class DispatcherService {
+    *http:Service;
+    private map<GenericServiceType> services = {};
+    private handler:NativeHandler nativeHandler = new ();
+
+    private ListenerConfig listenerConfig;
+
+    public function init(ListenerConfig listenerConfig) {
+        self.listenerConfig = listenerConfig;
+    }
+
+    isolated function addServiceRef(string serviceType, GenericServiceType genericService) returns error? {
+        if (self.services.hasKey(serviceType)) {
+            return error("Service of type " + serviceType + " has already been attached");
+        }
+        self.services[serviceType] = genericService;
+    }
+
+    isolated function removeServiceRef(string serviceType) returns error? {
+        if (!self.services.hasKey(serviceType)) {
+            return error("Cannot detach the service of type " + serviceType + ". Service has not been attached to the listener before");
+        }
+        _ = self.services.remove(serviceType);
+    }
+
+    // We are not using the (@http:payload GenericEventWrapperEvent g) notation because of a bug in Ballerina.
+    // Issue: https://github.com/ballerina-platform/ballerina-lang/issues/32859
+    resource function post .(http:Caller caller, http:Request request) returns http:Response|error? {
+        json payload = check request.getJsonPayload();
+        json[] eventsPayload = check payload.ensureType();
+        http:Response response = new;
+
+        error? validateRequestResult = self.validateRequest(request, payload);
+        if validateRequestResult is error {
+            string errorMessage = string `${eventValidationError} | ${validateRequestResult.message()}`;
+            log:printError(errorMessage);
+            response.statusCode = http:STATUS_NOT_ACCEPTABLE;
+            response.setPayload(errorMessage);
+            return response;
+        }
+
+        // Iterate event payload and emit each event
+        foreach var event in eventsPayload {
+            GenericDataType genericDataType = check event.cloneWithType(GenericDataType);
+            error? matchRemoteFuncResult = self.matchRemoteFunc(genericDataType);
+            if matchRemoteFuncResult is error {
+                WebhookEvent failedEvent = check genericDataType.ensureType(WebhookEvent);
+                log:printError(string `Error processing the event. EventID: ${<string>failedEvent.eventId}`);
```

Comment:
Shall we print the error too?.

Reply:
![image](https://user-images.githubusercontent.com/19235998/180128590-4938b847-a286-421f-9622-435a3ec2d0ee.png)

Added whole error log too.